### PR TITLE
MM-13934 - Plugin helper function `EnsureBot` doesn't compare bot details

### LINKED ISSUE #13934: Plugin helper function `EnsureBot` doesn't compare bot details
#### Summary
Plugin helper function `EnsureBot` doesn't compare bot details and only checks if a bot already exists for the plugin. This causes `EnsureBot` to return invalid bot data when, for example, changing a bot username or trying to create multiple bots.

#### Steps to reproduce
1. Create a plugin that uses `EnsureBot` to create a bot with username `foo`.
1. Install and run this plugin to create the bot user `foo`.
1. Update the plugin and replace `foo` with `bar` in `EnsureBot`. Re-run the updated plugin.
1. Pring the bot ID and check which bot user does it correspond to.

#### Expected behavior
On running `EnsureBot` with an updated username, a new bot must be created with updated details and the corresponding ID be returned.

#### Observed behavior (that appears unintentional) 
The existing bot `foo`'s ID is returned and no new bot user is created.

#### Possible fixes
The problem is at this line - https://github.com/mattermost/mattermost-server/blob/0e1a9f7e530061cdd2c7c17899e458afe2c83a9b/plugin/helpers_bots.go#L234

Instead of merely checking the presence of some bot, it should verify the username or other items of the bot and ensure the requested bot is created.

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Updated `ensureBot` plugin helper to updated found bot to match requested bot.

#### Ticket Link
Fixes #13934

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @harshilsharma for the contribution! Can you help look at the build failure?

File: plugin/helpers_bots.go [hunk 1]
```diff
@@ -231,9 +231,21 @@ func (p *HelpersImpl) ensureBot(bot *model.Bot) (retBotID string, retErr error)
...
+
+		// ensure existing bot is synced with what is being created
+		botPatch := &model.BotPatch{
+			Username:    &bot.Username,
+			DisplayName: &bot.DisplayName,
+			Description: &bot.Description,
+		}
+
+		if _, patchBotErr := p.API.PatchBot(botID, botPatch); patchBotErr != nil {
+			return "", errors.Wrap(patchBotErr, "failed to patch bot")
```
Comment (hanzei) [inline plugin/helpers_bots.go]: Given that the context of the `error` is clear, does it make sense to just go with the standard naming convention? 
```suggestion
		if _, err := p.API.PatchBot(botID, botPatch); err != nil {
			return "", errors.Wrap(err, "failed to patch bot")
```

Comment (harshilsharma63) [reply plugin/helpers_bots.go]: @hanzei yes, that makes sense. Fixed!

File: plugin/helpers_bots_test.go [hunk 2]
```diff
@@ -169,6 +189,37 @@ func TestEnsureBot(t *testing.T) {
...
+
+			p := &plugin.HelpersImpl{}
+			p.API = api
+
+			updatedTestbot := &model.Bot{
+				Username:    "updated_testbot",
+				DisplayName: "Updated Test Bot",
+				Description: "updated testbotdescription",
+			}
+			botID, err := p.EnsureBot(updatedTestbot)
```
Comment (ali-farooq0) [inline plugin/helpers_bots_test.go]: lets also assert the images as well, better to do
`botID, err := p.EnsureBot(testbot, plugin.ProfileImagePath(testImage), plugin.IconImagePath(testImage))`

along with
```
			api.On("SetProfileImage", expectedBotID, imageBytes).Return(nil)
			api.On("SetBotIconImage", expectedBotID, imageBytes).Return(nil)
```

Comment (harshilsharma63) [reply plugin/helpers_bots_test.go]: @ali-farooq0 I'll update the test this week.

**Comment by harshilsharma63** [comment]
@ali-farooq0 I'll update the test this week.

File: plugin/helpers_bots_test.go (hunk 2 above)
Comment (harshilsharma63) [reply plugin/helpers_bots_test.go]: Done.

**Comment by harshilsharma63** [comment]
@ali-farooq0 I've updated the test.

**Comment by ali-farooq0** [comment]
@hanzei do you who'll be doing QA for this?

**Comment by hanzei** [comment]
@prapti Would be the right one, as she also reviewed https://github.com/mattermost/mattermost-server/pull/12153.

@prapti Let me know if you have any questions.

**Comment by hanzei** [comment]
@prapti Gentle reminder to review this PR. Let me know if I should assign someone else as QA review.

**Comment by teresa-novoa** [comment]
> @prapti Gentle reminder to review this PR. Let me know if I should assign someone else as QA review.

**Comment by saturninoabril** [comment]
@ali-farooq0 Do we have a corresponding Jira ticket for this PR and what would be its milestone?

**Comment by ali-farooq0** [comment]
@saturninoabril nope, no jira ticket. We can just target this for the next release (master)

Is this PR good to go and can be merged?

**Comment by saturninoabril** [comment]
Got it and yes, good to merge. Thanks @ali-farooq0

**Comment by teresa-novoa** [comment]
@ali-farooq0 merging is blocked, can you help with it? _ee/circleci — Failed checking the translations._ Thanks!

**Comment by hanzei** [comment]
Thanks for the fix @harshilsharma63 :tada:

**Pull request merged**